arangod/RocksDBEngine/RocksDBReplicationContext.cpp

```diff
@@ -266,17 +327,69 @@ Result RocksDBReplicationContext::getInventory(TRI_vocbase_t& vocbase, bool incl
     return true;
   };
 
-  TRI_ASSERT(_snapshot != nullptr);
-  // FIXME is it technically correct to include newly created collections ?
-  // simon: I think no, but this has been the behavior since 3.2
-  TRI_voc_tick_t tick = TRI_NewTickServer();  // = _lastArangoTick
+  TRI_voc_tick_t tick = TRI_NewTickServer();
+  VPackBuilder inventory;
+
   if (global) {
     // global inventory
-    vocbase.server().getFeature<DatabaseFeature>().inventory(result, tick, nameFilter);
+    vocbase.server().getFeature<DatabaseFeature>().inventory(inventory, tick, nameFilter);
```

Comment:
Okay to keep it for now.
But this looks like it could be done simpler, by not serializing / deserializing the inventory into a VelocyPack but apply the below callback right on the in memory structure of the inventory.

Reply:
E.g. hand the callback in via a similar variant of this method here, instead of the VelocyPack result.

Reply:
Yes. For now, I wanted to keep the existing functionality of `inventory` without duplicating its internal code. To keep changes limited to the one call site and not all users of `inventory`. We can prettify this later in devel if we want to.